### PR TITLE
Improve Bloom performance

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Represent internal buffer as fixed size buffer rather than array
- Vectorise `Accumulate` directly over the fixed size buffer
- Improve asm produced for `Set`/`Get`/`GetExtract`
- Change `BloomExtract` to a `readonly struct` and pass via `in`

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by benaadams** [comment]
Is dominated by Keccak calc so only saves 2%

|              Method |     Mean |   Error |  StdDev |
|-------------------- |---------:|--------:|--------:|
| ReconstructFromLogs (Old) | 965.2 us | 5.79 us | 8.66 us |
| ReconstructFromLogs (New) | 953.8 us | 8.62 us | 12.64 us |

File: src/Nethermind/Nethermind.Core/Bloom.cs [hunk 1]
```diff
@@ -108,86 +129,107 @@ public override bool Equals(object? obj)
...
         public void Accumulate(Bloom bloom)
         {
-            Bytes.AsSpan().Or(bloom.Bytes);
+            ref BloomData data = ref bloom._data;
+            for (int i = 0; i < Bloom.Vector256Length; i++)
+            {
+                Vector256<byte> v1 = Vector256.LoadUnsafe(ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref _data), i * Bloom.Vector256Size));
+                Vector256<byte> v2 = Vector256.LoadUnsafe(ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref data), i * Bloom.Vector256Size));
+                Vector256.StoreUnsafe(Vector256.BitwiseOr(v1, v2), ref Unsafe.AddByteOffset(ref Unsafe.As<BloomData, byte>(ref _data), i * Bloom.Vector256Size));
+            }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Bloom.cs:169]: This already was using Vector256 underneath.

**Review by LukaszRozmej — APPROVED** [review]
Quite the micro optimization.

File: src/Nethermind/Nethermind.Core/Bloom.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Bloom.cs:169]: Yeah but variable length with via span loading and only for x64 `Vector256.BitwiseOr` also works for Arm. Ends up with a ton more instructions

**Comment by benaadams** [comment]
> Quite the micro optimization. Variable length blooms would be much more useful.

I was hoping for much more 😢 Keccak is the gorilla in the room; I have some partial vectorising reworks that I might revisit in light of this result

**Pull request closed without merging**